core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -87,6 +113,8 @@ class StorageStatusListener extends SparkListener {
   override def onBlockManagerRemoved(blockManagerRemoved: SparkListenerBlockManagerRemoved) {
     synchronized {
       val executorId = blockManagerRemoved.blockManagerId.executorId
+      removedExecutorIdToStorageStatus.put(executorId,
+          executorIdToStorageStatus.get(executorId).get)
```

Comment:
I really think it would be nice to keep the time the executor was removed as well from `blockManagerRemoved.time` to show in the UI